src/libstore/realisation.hh

```diff
@@ -3,6 +3,34 @@
 #include "path.hh"
 #include <nlohmann/json_fwd.hpp>
 
+
+/* Awfull hacky generation of the comparison operators by doing a lexicographic
+ * comparison between the choosen fields
+ * ```
+ * GENERATE_CMP(ClassName, my->field1, my->field2, ...)
+ * ```
+ *
+ * will generate comparison operators semantically equivalent to:
+ * ```
+ * bool operator<(const ClassName& other) {
+ *   return field1 < other.field1 && field2 < other.field2 && ...;
+ * }
+ * ```
+ */
+#define GENERATE_ONE_CMP(COMPARATOR, MY_TYPE, FIELDS...) \
```

Comment:
Maybe move this into a separate file in libutil?

Reply:
OK, I guess that can be useful elsewhere.

Done in 343687405d6de18c0b1b7aab814e3ea6a0a61420